src/modm/container/dynamic_array.hpp

```diff
@@ -258,7 +259,7 @@ namespace modm
 		/**
 		 * \brief	Forward iterator
 		 */
-		class iterator
+		class iterator : public std::iterator<std::forward_iterator_tag, T>
```

Comment:
It would make sense to have a random access iterator here, but that would require extending the iterator class with operators `[]`, `+`, `-`, `+=`, `-=`. Otherwise things like std::sort don't work which need random access.

These container implementations were originally written for xpcc before we had a standard library on all platforms and when there was no C++11. Now modm uses the standard library which provides vector, list and forward_list. I don't see why someone should prefer the modm equivalents for these three cases. I'm not sure it's worth the effort to fix them.
@salkinium We should consider to deprecate or remove the containers the standard library already provides.

Reply:
I would strongly prefer either using the std libc++ and/or integrating a third party one [like ETL](https://www.etlcpp.com/). Definitely not interested in fixing/maintaining our containers, that's something that other libs can do way better.

Reply:
Is there a specific reason for using these containers instead of the standard ones, @mikewolfram? Avoiding heap? Size/efficiency concerns? What do these containers do better?

Reply:
Well, since I expected those containers to be used in modm, I wanted to use them as well for keeping it simple.

I've been already looking into ETL and was thinking about using some of it.